subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/Stripper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.toolchain.internal;
+
+import com.google.common.collect.Lists;
+import org.gradle.api.Action;
+import org.gradle.internal.operations.BuildOperationExecutor;
+import org.gradle.internal.operations.BuildOperationQueue;
+import org.gradle.internal.work.WorkerLeaseService;
+import org.gradle.nativeplatform.internal.StripperSpec;
+
+import java.io.File;
+import java.util.List;
+
+public class Stripper extends AbstractCompiler<StripperSpec> {
+    public Stripper(BuildOperationExecutor buildOperationExecutor, CommandLineToolInvocationWorker commandLineToolInvocationWorker, CommandLineToolContext invocationContext, WorkerLeaseService workerLeaseService) {
```

Comment:
The big difference here is Xcode doesn't strip the symbol from the binary. It feels like stripping should be something that happens when installing the application or preparing for packaging. Let's talk about this.

Reply:
Xcode has a build activity and a deploy activity.  Build compiles/links the binary, while deploy does the extract/strip.  When you run/debug in Xcode it only invokes the build activity.  We do the same thing by pointing the build activity to the `linkRelease` task (which only produces the unstripped binary).  However, upstream libraries will link against the stripped binaries (see discussion above).